clients/src/main/java/org/apache/kafka/common/requests/ReadShareGroupStateResponse.java

```diff
@@ -65,4 +68,50 @@ public static ReadShareGroupStateResponse parse(ByteBuffer buffer, short version
         new ReadShareGroupStateResponseData(new ByteBufferAccessor(buffer), version)
     );
   }
+
+  public static ReadShareGroupStateResponseData getSuccessResponse(
+      Uuid topicId,
+      int partition,
+      long startOffset,
+      int stateEpoch,
+      List<ReadShareGroupStateResponseData.StateBatch> stateBatches
+  ) {
+    return new ReadShareGroupStateResponseData()
+        .setResults(Collections.singletonList(
+            new ReadShareGroupStateResponseData.ReadStateResult()
+                .setTopicId(topicId)
+                .setPartitions(Collections.singletonList(
+                    new ReadShareGroupStateResponseData.PartitionResult()
+                        .setErrorCode(Errors.NONE.code())
```

Comment:
The error code defaults to 0. The error message defaults to null (which is more efficient than an empty string). I would not bother setting either of them here.